pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -109,17 +109,32 @@ func (c *client) IsFunctionExist(ctx context.Context, name string) (bool, error)
 
 func (c *client) CreateFunction(ctx context.Context, fm FunctionManifest) error {
 	input := &lambda.CreateFunctionInput{
-		Code: &types.FunctionCode{
-			ImageUri: aws.String(fm.Spec.ImageURI),
-		},
-		PackageType:  types.PackageType("Image"),
-		Role:         aws.String(fm.Spec.Role),
 		FunctionName: aws.String(fm.Spec.Name),
+		Role:         aws.String(fm.Spec.Role),
+		MemorySize:   aws.Int32(fm.Spec.Memory),
+		Timeout:      aws.Int32(fm.Spec.Timeout),
 		Tags:         fm.Spec.Tags,
 		Environment: &types.Environment{
 			Variables: fm.Spec.Environments,
 		},
 	}
+	// Container image packing.
+	if fm.Spec.ImageURI != "" {
+		input.PackageType = types.PackageTypeImage
+		input.Code = &types.FunctionCode{
+			ImageUri: aws.String(fm.Spec.ImageURI),
+		}
+	}
+	// Zip packing which stored in s3.
+	if fm.Spec.S3Bucket != "" {
+		input.PackageType = types.PackageTypeZip
+		input.Code = &types.FunctionCode{
+			S3Bucket:        aws.String(fm.Spec.S3Bucket),
+			S3Key:           aws.String(fm.Spec.S3Key),
+			S3ObjectVersion: aws.String(fm.Spec.S3ObjectVersion),
```

Comment:
I want to ask about this object version. Is it configurable or predictable from the developer?
I mean our designed workflow is as
1. developer runs CI tasks to upload their source code to their S3 bucket 
2. developer updates (or by our Event Watcher) `s3ObjectVersion` field in function.yaml file to deploy

At the first step, can they easily know the `s3ObjectVersion` of the newly uploaded data?
The version can be specified or generated automatically with a random string?

Reply:
> The version can be specified or generated automatically with a random string?
 
 It will be generated by S3 due to this [docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html)
 
 > can they easily know the s3ObjectVersion of the newly uploaded data?
 
 At the CI step, the `sam package ...` command which does (1) as you mentioned will return required value to detect where to retrieve the object (key, version).

Reply:
Besides, one more thing to mention is that the `s3Objectversion` is not a required field, in case users do not enable versioning feature on s3, they still can upload their zip function code to the s3 and use only s3Bucket and s3Key (the path to object) to deploy their lambda function.